DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py

```diff
@@ -0,0 +1,169 @@
+
+    await IpLink.set(input_data=[{device: [{"device": "br0", "operstate": "up"}]}])
+    assert out[0][device]["rc"] == 0, f"Failed setting bridge to state UP.\n {out}"
+
+    await IpLink.add(input_data=[{device: [{"device": port} for port in ports]}])
+    assert out[0][device]["rc"] == 0, f"Failed creating link\n {out}"
+
+    out = await IpLink.set(input_data=[{device: [{"device": port, "operstate": "up"} for port in ports]}])
+    assert out[0][device]["rc"] == 0, f"Failed setting link to state UP\n {out}"
+
+    await IpLink.set(input_data=[{device: [{"device": port, "master": "br0"} for port in ports]}])
+    assert out[0][device]["rc"] == 0, f"Failed to enslave link(s) to bridge.\n {out}"
+
+    # 2.Set links to vlans.
+    for vlan, setting in vlans.items():
+        pvid = True if setting["pvid"] is True else ""
+        is_untagged = True if setting["untagged"] is True else ""
+        for port in setting["ports"]:
+            out = await BridgeVlan.add(input_data=[
+                {device: [{"device": f"{ports[port - 1]}", "vid": int(vlan), "pvid": pvid, "untagged": is_untagged}]}])
+            assert out[0][device]["rc"] == 0, f"Vlan: {vlan}  is not added to link: {ports[port - 1]}\n {out}"
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24},
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24},
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24},
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    # 3. Map receiving and non receiving ports.
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"], dev_groups[tg_ports[2]][0]["name"],
+                dev_groups[tg_ports[3]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {}
+    for vlan in vlans:
+        vlan = vlan if vlan != "X" else ""
+        streams.update(
+            {f"traffic with VLAN ID : {vlan}": {
+                "type": "raw",
+                "ip_source": tx_ports,
+                "ip_destination": rx_ports,
+                "srcMac": "02:00:00:00:00:AA",
+                "dstMac": "ff:ff:ff:ff:ff:ff",
+                "vlanID": vlan,
+            }}, )
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(20)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+
+    # 5.1 Verify traffic with vlan "x" received on port 4
+    for row in stats.Rows:
+        if "X" in row["Traffic Item"] and dev_groups.keys()[3] in row["Rx Port"]:
+            assert tgen_utils_get_loss(row) == 0.000
```

Comment:
Please add here and bellow assert messages

Reply:
reworked